pandas/io/parsers.py

```diff
@@ -2435,6 +2435,21 @@ def _clean_mapping(mapping):
             clean_na_values = self.na_values
             clean_na_fvalues = self.na_fvalues
 
+        try:
+            if isinstance(clean_dtypes, dict):
+                for col, dt in clean_dtypes.items():
+                    if is_bool_dtype(dt) and data[col][data[col] == ''].size:
+                        raise ValueError("Bool column has NA values in "
+                                         "column {column}".format(column=col))
+            elif (isinstance(clean_dtypes, string_types) and
+                  is_bool_dtype(clean_dtypes)):
+                for col, values in data.items():
+                    if any(isna(values)):
+                        raise ValueError("Bool column has NA values in "
+                                         "column {column}".format(column=col))
+        except (AttributeError, TypeError):  # invalid input to is_bool_dtype
+            pass
+
```

Comment:
I thought about this some more, and I realized that my suggestion was not a good one, as it wouldn't allow us to check for values that have been designated as `NaN` i.e. if a row has `[True, "NAN", False]`, with "NAN" considered a null value, attempting to cast to boolean should also raise.

Can you see if you can move this logic check to `self._convert_to_ndarrays` ?  Sorry about the jerking around here...